### PR TITLE
[feature] Handle schema reset for mongo

### PR DESCRIPTION
**Pull request by nikophil**
Hello,

here is a proposition to support the `DatabaseResetter` with mongo.

I've ran into few problems for which I've made some trade-off, see my comments in the code.
I'm not quite happy with the implementation because `self::isDAMADoctrineTestBundleEnabled()` pops in every layers (`ResetDatabase` trait, `DatabaseResetter` and `ORMDatabaseResetter`) althought it is only really needed for orm...

I've not created automated tests for new classes yet: I'd rather wait we're OK on an implementation (I don"t even know if it's necessary since these class will be used into the test suite). 
I've also deactivated codecoverage because it resulted in too much noise in code review

### CODE REVIEW

File: src/ChainManagerRegistry.php
Comment (nikophil) [inline src/ChainManagerRegistry.php:54]: `ManagerRegistry::getManagers()` actually return `array<string, ObjectManager>`, where the key is the objectManager's name. So there is a risk of overlay.

This actually makes our method return `list<ObjectManager>` but the interface says `@return ObjectManager[]` so we don't break the contract...

File: src/Test/DatabaseResetter.php
Comment (nikophil) [inline src/Test/DatabaseResetter.php:54]: This is where we have a problem with odm & orm & dama: if we have a global state for odm, it wouldn't be persisted.

File: src/Test/DatabaseResetter.php
Comment (nikophil) [inline src/Test/DatabaseResetter.php:37]: There is no such things like `doctrine:database:drop` or `doctrine:database:create` in `doctrine/odm`. 
That's why we only use ORM here, and why the interface I've created only refers to "reset schema"

File: src/Test/ODMSchemaResetter.php
Comment (nikophil) [inline src/Test/ODMSchemaResetter.php:50]: for now, we only deal with `default` manager, but we can imagine to introduce new `FOUNDRY_RESET_ODM_MANAGERS` env var

File: src/Test/ResetDatabase.php
Comment (nikophil) [inline src/Test/ResetDatabase.php:56]: Now even with dama, the kernel will be booted: we need to know whether or not we have odm in the kernel, this condition has been moved into `ORMDatabaseResetter::resetSchame()`

File: src/Test/SchemaResetterInterface.php
Comment (kbond) [inline src/Test/SchemaResetterInterface.php:8]: Since this is an internal interface, let's just make it an abstract class (`SchemaResetter`) that includes the helper method from the `CommandRunner` trait and remove the trait.

**Review by kbond — CHANGES_REQUESTED** [review]
Nice work! I did a quick review and I think there are some opportunities to cleanup/refactor but I'd rather wait until we have odm tests. This will allow us to refactor with confidence.

Once we have new, passing tests, I'll merge into the primary mongo-support branch and we can iterate on it with additional PRs.

**Comment by kbond** [comment]
> I'm not quite happy with the implementation because self::isDAMADoctrineTestBundleEnabled() pops in every layers

Fine for now I think, we may be able to refactor away on a followup iteration.

File: .github/workflows/ci.yml
Comment (nikophil) [inline .github/workflows/ci.yml:135]: maybe it's time to use a matrix for all test cases?

Comment (kbond) [reply .github/workflows/ci.yml:135]: Let's save for a future PR.

**Review by kbond — CHANGES_REQUESTED** [review]
Looks good, couple comments/suggestions:
1. Can we just check `DATABASE_URL`/`MONGO_URL` to determine if they are enabled? (and remove `USE_ORM`/`USE_ODM`).
2. Can you disable psalm for now (and remove any psalm related comments added). We'll do a separate PR after to re-enable and fix.

**Review by kbond — APPROVED** [review]
LGTM, when you're ready, unmark as draft and I'll merge into the `mongo-support` branch.

**Comment by nikophil** [comment]
hi @kbond 
I think the PR is ready

**Pull request merged**
